main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -649,7 +655,7 @@ object BuildServerProtocol {
   private def scalaMainClassesTask: Initialize[Task[ScalaMainClassesItem]] = Def.task {
     val jvmOptions = Keys.javaOptions.value.toVector
     val mainClasses = Keys.discoveredMainClasses.value.map(
-      ScalaMainClass(_, Vector(), jvmOptions)
+      ScalaMainClass(_, Vector(), jvmOptions, Vector.empty)
```

Comment:
We should probably fill the `environmentVariables` field with `envVars.value` here as well.

Reply:
Yes, I had the same feeling first, but then decided to give it a different meaning. As the origin of `ScalaMainClass` data is external to sbt process. (metals or something else) and the data is restored from serialized object normally - the client will not be aware what settings are actual inside sbt. So, I treat the data incoming in this place as "additional" environment variables.

https://github.com/sbt/sbt/pull/6397/files#diff-2ad23bc2de451984eb3983aceb2f15262cd8748b845faebae937a4d8f762f244R591
They are then concatenated with ones from sbt process, when the execution is actually initiated.

But this assumption of mine could be wrong, because I didn't spend enough time with the codebase.

Reply:
I agree with you that the incoming environment variables should be treated as additionals. This is not how the `jvmOptions` are treated but maybe it was a mistake.

However, this method (`scalaMainClassesTask`) is the implementation of the `buildTarget/scalaMainClasses` request sent by the client to the server (sbt) to learn about the project main classes and how they are configured. So it is probably safe and better to not hide the `envVars`, even if the client do not use them later.

Reply:
For instance, the client might want to run the `scalaMainClass` by itself (not through sbt), in such case it would like to have the `envVars` are they are configured in sbt.

Reply:
Read the code once again. Now I understand what you suggested. Thank you for the explanation @adpi2!

I added commit, where I moved concatenation to parts, where we are instantiating `ScalaMainClass` - either recovering serialized version or creating with default constructor. And removed unnecessary concatenation in the moment, when the forked process is invoked.